### ISSUE TITLE
If you use the following command, it seems a  **binary accuracy** command.

### ISSUE THREAD

**Issue by 245789064** [shamanez/BERT-like-is-All-You-Need#20]
If you use the following command, it seems a  **binary accuracy** command.

`python train.py --data ./T_data/iemocap ---restore-file None --task emotion_prediction --reset-optimizer --reset-dataloader --reset-meters --init-token 0 --separator-token 2 --arch robertEMO_large --criterion emotion_prediction_cri --num-classes 8 --dropout 0.1 --attention-dropout 0.1 --weight-decay 0.1 --optimizer adam --adam-betas "(0.9, 0.98)" --adam-eps 1e-06 --clip-norm 0.0 --lr-scheduler polynomial_decay --lr 1e-05 --total-num-update 2760 --warmup-updates 165 --max-epoch 10 --best-checkpoint-metric loss --encoder-attention-heads 2 --batch-size 1 --encoder-layers-cross 1 --no-epoch-checkpoints --update-freq 8 --find-unused-parameters --ddp-backend=no_c10d --binary-target-iemocap --a-only --t-only --pooler-dropout 0.1 --log-interval 1 --data-raw ./iemocap_data/`

Because when I use the following **validation command** to predict emotions, preds may be **[0, 0, 0, 0]** and **[1, 0, 1, 0]** in line 113, **fairseq/criterions/emotion_prediction_cri.py**.

`python validate.py --data ./T_data/iemocap --path './checkpoints/checkpoint_best.pt' --task emotion_prediction --valid-subset test --batch-size 1`

```
            logits=logits.view(-1, 2)
            targets= targets.view(-1)

            loss = F.nll_loss(
                F.log_softmax(logits, dim=-1, dtype=torch.float32),
                targets,
                reduction='sum',
            )

            preds = logits.max(dim=1)[1]

            ncorrect=(preds == targets).sum().item()
```

_Originally posted by @Chenpuh in https://github.com/shamanez/BERT-like-is-All-You-Need/issues/18#issuecomment-822376133_

**Comment by 245789064** [comment]
I have the same problem,,

**Comment by 245789064** [comment]
so,how can I get the 4- class acc
